distributed/tests/test_cancelled_state.py

```diff
@@ -452,3 +454,113 @@ async def get_data(self, comm, *args, **kwargs):
     # w1 closed
 
     assert await f4 == 6
+
+
+@pytest.mark.parametrize("wait_for_processing", [True, False])
+@pytest.mark.parametrize("raise_error", [True, False])
+@gen_cluster(client=True, timeout=3)
+async def test_resumed_cancelled_handle_compute(
+    c, s, a, b, raise_error, wait_for_processing
+):
```

Comment:
This covers the change sin handle_compute where I moved around task states in the switch statement and added an error handler that was missing before